WordPress/src/main/java/org/wordpress/android/ui/news/NewsManager.kt

```diff
@@ -0,0 +1,52 @@
+package org.wordpress.android.ui.news
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MediatorLiveData
+import org.wordpress.android.models.news.NewsItem
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Business logic related to fetching/showing the News Card (a card used for announcing new features/updates).
+ */
+@Singleton
+class NewsManager @Inject constructor(val newsService: NewsService, val appPrefs: AppPrefsWrapper) {
+    private val dataSourceMediator: MediatorLiveData<NewsItem?> = MediatorLiveData()
+    private var localServiceDataSource: LiveData<NewsItem?>? = null
+
+    fun newsItemSource(): LiveData<NewsItem?> {
+        if (localServiceDataSource == null) {
+            localServiceDataSource = newsService.newsItemSource()
+            dataSourceMediator.addSource(localServiceDataSource as LiveData<NewsItem?>) {
+                if (shouldPropagateToUI(it)) {
+                    dataSourceMediator.value = it
+                }
+            }
+        }
+        return dataSourceMediator
+    }
+
+    fun dismiss() {
```

Comment:
what do you think about passing the dismissed version as a parameter here? I can imagine a race condition of updating the card at the same moment as dismissing it.

Reply:
👍